pkg/volume/nfs/nfs.go

```diff
@@ -285,6 +285,12 @@ func (nfsMounter *nfsMounter) SetUpAt(dir string, mounterArgs volume.MounterArgs
 		os.Remove(dir)
 		return err
 	}
+
+	if !nfsMounter.readOnly {
+		volume.SetVolumeOwnership(nfsMounter, mounterArgs.FsGroup, mounterArgs.FSGroupChangePolicy, util.FSGroupCompleteHook(nfsMounter.plugin, nil))
```

Comment:
I'm not sure we can add this without potentially breaking existing users that may have been unintentionally depending on the old behavior.

If this is needed, consider using the csi-driver-nfs plugin instead in combination with https://kubernetes-csi.github.io/docs/support-fsgroup.html

Reply:
If https://github.com/kubernetes-csi/csi-driver-nfs is suggested to use, I prefer to /hold or /close this PR as it has some behavior change.

I go through some other intree volume driver code, `volume.SetVolumeOwnership` is called in some of them for not read only scanerios and some check if the plugin supports fsType.

 #95945

Reply:
We don't support fsGroup on NFS because some NFS servers runs with `root_squash` and kubelet is not able to change ownership of files there.

In CSI, the CSI driver vendor / cluster admin can set up `CSIDriver.Spec.FSGroupPolicy` to indicate if Kubernetes should do the ownership change or not. I would leave the in-tree code as it is, this PR will break at least some deployments.
/hold